src/lib/wapi.js

```diff
@@ -448,9 +448,9 @@ window.WAPI.sendMessageWithThumb = function (thumb, url, title, description, cha
         description: description,
         matchedText: url,
         title: title,
-        thumbnail: thumb
+        thumbnail: thumb // Thumbnail max size allowed: 200x200
     };
-    chatSend.sendMessage(url, { linkPreview: linkPreview, mentionedJidList: [], quotedMsg: null, quotedMsgAdminGroupJid: null });
+    chatSend.sendMessage(`${url} ${text}`, { linkPreview: linkPreview, mentionedJidList: [], quotedMsg: null, quotedMsgAdminGroupJid: null });
```

Comment:
How about some sort of url check? If the URL exists in the text then leave it as is otherwise use this format.

```javascript
chatSend.sendMessage(text.includes(url) ? text : `${url}\n${text}`, ...
```